purchase_work_acceptance_evaluation/report/work_acceptance_evaluation_report.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, tools
+
+
+class WorkAcceptanceEvaluationReport(models.Model):
+    _name = "work.acceptance.evaluation.report"
+    _description = "Work Acceptance Evaluation Report"
+    _order = "partner_id,date_accept"
+    _auto = False
+
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Vendor",
+        index=True,
+    )
+    wa_id = fields.Many2one(
+        comodel_name="work.acceptance",
+        string="WA Reference",
+        index=True,
+    )
+    date_accept = fields.Datetime(
+        string="Accepted Date",
+    )
+    case_id = fields.Many2one(
+        comodel_name="work.acceptance.evaluation",
+        string="Case Name",
+        index=True,
+    )
+    score_id = fields.Many2one(
+        comodel_name="work.acceptance.evaluation.score",
+        string="Score",
+        index=True,
+    )
+    note = fields.Char(
+        string="Note",
+    )
+
+    def init(self):
```

Comment:
Please use this method, this allow use to make report from query (no view need). And I find it more flexible in many ways.

```
    @property
    def _table_query(self):
        return '%s %s %s' % (self._select(), self._from(), self._where())

```

Reply:
Fixed, Thanks for your suggest.

May I ask, this function is create view on the database like a query "create or replace view ...." ?

Reply:
No there will be no view. It will execute query on the fly.

See sample in account_invoice_report.py